pkg/config/load.go

```
@@ -373,41 +378,64 @@ func processNode(node *yaml.Node, v *viper.Viper, currentPath string) {
 	// Process children nodes (for sequences/lists)
 	for _, child := range node.Content {
-		processNode(child, v, currentPath)
+		if err := processNode(child, v, currentPath); err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
-func processScalarNode(node *yaml.Node, v *viper.Viper, currentPath string) {
+func processScalarNode(node *yaml.Node, v *viper.Viper, currentPath string) error {
```

golangci-lint (1.64.8)

[error] cognitive complexity 26 of func `processScalarNode` is high (> 20)

(gocognit)